src/lib/yacc.rs

```diff
@@ -57,28 +59,41 @@ lazy_static! {
     static ref RE_TERMINAL: Regex = {
         Regex::new("^(?:(\".+?\")|('.+?')|([a-zA-Z_][a-zA-Z_0-9]*))").unwrap()
     };
-    static ref RE_WHITESPACE: Regex = {
-        Regex::new(r"^\s+").unwrap()
-    };
 }
 
 /// The actual parser is intended to be entirely opaque from outside users.
 impl YaccParser {
     fn new(src: String) -> YaccParser {
         YaccParser {
-            src    : src,
-            grammar: GrammarAST::new()
+            src     : src,
+            newlines: vec![],
+            grammar : GrammarAST::new()
         }
     }
 
+    fn off_to_line_col(&self, off: usize) -> (usize, usize) {
+        if self.newlines.len() == 0 {
+            return (1, off + 1);
+        }
+        for (line_m1, &line_off) in self.newlines.iter().enumerate() {
+            if line_off >= off {
+                return (line_m1 + 1, off - line_off + 1);
+            }
+        }
+        (self.newlines.len() + 1, off - self.newlines[self.newlines.len() - 1])
```

Comment:
Suggestion: Use ``self.newlines.last().unwrap()`` instead of ``self.newlines[self.newlines.len() - 1]``